inference/models/owlv2/owlv2.py

```diff
@@ -196,6 +200,58 @@ def make_class_map(
     return class_map, class_names
 
 
+def hash_function(value: Any) -> Hash:
+    # wrapper so we can change the hashing function in the future
+    return hashlib.sha1(value).hexdigest()
```

Comment:
If I understand correctly, this moves from sha256 to sha1 - sha1 is [deprecated according to NIST](https://www.nist.gov/news-events/news/2022/12/nist-retires-sha-1-cryptographic-algorithm), so we should treat it as such IMO.

Reply:
it is deprecated for secure applications, but this isn't a secure application .. only handling potential collisions between up to 1000 entries, and probability of an issue is vanishingly small .. sha1 is some >2x faster than sha256 and I am prioritizing speed

if you think we should prioritize using a secure hash, we could integrate a (much faster) secure hash via a third party library? I just don't want hashing the images to be the slowest part of the pipeline

Reply:
what I mean by broken for secure applications is that, if someone wants to find a collision they can do so with >10k$ of GPU https://sha-mbles.github.io/ which I think is more than is likely to be an issue for our case! :)